### PR TITLE
Security 1 object model

### PR DESCRIPTION
**Pull request by KacFlor**

### CODE REVIEW

File: src/main/java/com/KacFlor/ShopSpring/SpringShop.java [hunk 1]
```diff
@@ -6,9 +6,9 @@
 
 @SpringBootApplication
 @RestController
-public class SpringShop {
+public class SpringShop{
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/SpringShop.java:9]: I assume you applied formatting here, yes?

File: src/main/java/com/KacFlor/ShopSpring/config/CardDataConfig.java [hunk 2]
```diff
@@ -9,13 +9,13 @@
...
-public class CardDataConfig {
+public class CardDataConfig{
 
-    public CardDataConfig() {
+    public CardDataConfig(){
     }
 
     @Bean
-    CommandLineRunner commandLineRunner14(CardDataRepository cardDataRepository) {
+    CommandLineRunner commandLineRunner14(CardDataRepository cardDataRepository){
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/config/CardDataConfig.java]: why 14?

Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/config/CardDataConfig.java]: Name it commandLineRunnerCardData if you rly want to use it, apply everywhere

File: src/main/java/com/KacFlor/ShopSpring/SpringShop.java (hunk 1 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/SpringShop.java:9]: Yes

File: src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import com.KacFlor.ShopSpring.security.RegisterRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api")
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java]: I would change this url to "authentication"

File: src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    @PostMapping("/register")
+    public ResponseEntity<AuthenticationResponse> register(
+            @RequestBody RegisterRequest request
+    ){
+        return ResponseEntity.ok(service.register(request));
+
+    }
+
+    @PostMapping("/authenticate")
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java]: I would change this url to "login", but if you want to leave it as it is then it is your decision

File: src/main/java/com/KacFlor/ShopSpring/controller/CardDataController.java [hunk 5]
```diff
@@ -13,17 +16,17 @@
 @RequestMapping(
         path = {"carddata"}
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/CardDataController.java]: Change "carddata" to "card-data". See https://restfulapi.net/resource-naming/

File: src/main/java/com/KacFlor/ShopSpring/controller/CategoryController.java [hunk 6]
```diff
@@ -5,6 +5,7 @@
 import com.KacFlor.ShopSpring.service.CategoryService;
 import com.KacFlor.ShopSpring.model.Category;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/CategoryController.java]: Is it needed?

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java [hunk 7]
```diff
@@ -6,6 +6,7 @@
 import com.KacFlor.ShopSpring.model.User;
 import com.KacFlor.ShopSpring.service.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: Is it needed?

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java [hunk 8]
```diff
@@ -14,16 +15,27 @@
...
     }
 
     @GetMapping
-    public List<User> getUser() {
-        return this.userService.getUser();}
+    public List<User> getUser(){
+        return this.userService.getUser();
+    }
+
+    @GetMapping("user")
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: This endpoint url is "user/user" here, this is as test right?

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java [hunk 9]
```diff
@@ -14,16 +15,27 @@
...
+    public String helloUser(){
+        return "Hello User";
+    }
+
+    @GetMapping("admin")
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: This endpoint url is "user/admin" here, this is as test right?

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java [hunk 10]
```diff
@@ -14,16 +15,27 @@
...
     @GetMapping
-    public List<User> getUser() {
-        return this.userService.getUser();}
+    public List<User> getUser(){
+        return this.userService.getUser();
+    }
+
+    @GetMapping("user")
+    public String helloUser(){
+        return "Hello User";
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: Is is automatically wrapped in json right?

File: src/main/java/com/KacFlor/ShopSpring/controller/WishlistController.java [hunk 11]
```diff
@@ -13,17 +13,17 @@
...
     @Autowired
-    public WishlistController(WishlistService wishlistService) {
+    public WishlistController(WishlistService wishlistService){
         this.wishlistService = wishlistService;
     }
 
     @GetMapping
-    public List<Wishlist> getWishlist() {
+    public List<Wishlist> getWishlist(){
         return this.wishlistService.getWishlist();
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/controller/WishlistController.java:27]: Is is automatically wrapped in json right?

File: src/main/java/com/KacFlor/ShopSpring/dao/UserRepository.java [hunk 12]
```diff
@@ -1,9 +1,14 @@
 package com.KacFlor.ShopSpring.dao;
 
 import com.KacFlor.ShopSpring.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/dao/UserRepository.java]: Is it needed?

File: src/main/java/com/KacFlor/ShopSpring/security/AuthenticationRequest.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class AuthenticationRequest{
+
+    private String login;
+
+    String password;
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/security/AuthenticationRequest.java]: Shouldn't it be private?

File: src/main/java/com/KacFlor/ShopSpring/service/AuthenticationService.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    private final UserRepository userRepository;
+
+    private final CustomerRepository customerRepository;
+
+    private final AuthenticationManager authenticationManager;
+
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    public AuthenticationResponse register(RegisterRequest request){
+        var customer = new Customer("Adam", "Florczyk", "kacper.florczyk@onet.pl", "staffa");
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/service/AuthenticationService.java]: These are default data? Maybe we should leave them empty?

File: src/main/java/com/KacFlor/ShopSpring/service/CardDataService.java [hunk 15]
```diff
@@ -3,20 +3,22 @@
 import com.KacFlor.ShopSpring.model.CardData;
 import com.KacFlor.ShopSpring.dao.CardDataRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.annotation.Secured;
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/service/CardDataService.java]: Is it needed?

File: src/main/java/com/KacFlor/ShopSpring/service/JwtService.java [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import static javax.crypto.Cipher.SECRET_KEY;
+
+@Service
+public class JwtService{
+
+    private String jwtSecret = "4261656C64756E67491624918624781263819203713128936123781631290312739123871290371238126312783o6123790p126389123612839o126312938126";
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/service/JwtService.java]: Is it written by hand xd? Maybe use some random text generator from web

File: src/main/java/com/KacFlor/ShopSpring/config/CardDataConfig.java (hunk 2 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/config/CardDataConfig.java]: Done

File: src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java (hunk 3 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java (hunk 4 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/AuthenticationController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/CardDataController.java (hunk 5 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/CardDataController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/CategoryController.java (hunk 6 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/CategoryController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 7 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 8 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 9 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 10 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/WishlistController.java (hunk 11 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/controller/WishlistController.java:27]: yes

File: src/main/java/com/KacFlor/ShopSpring/dao/UserRepository.java (hunk 12 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/dao/UserRepository.java]: done

File: src/main/java/com/KacFlor/ShopSpring/security/AuthenticationRequest.java (hunk 13 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/security/AuthenticationRequest.java]: done

File: src/main/java/com/KacFlor/ShopSpring/service/AuthenticationService.java (hunk 14 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/service/AuthenticationService.java]: done (i will make that after when user create a account ,he will be able to set customer data like name ,surname ,etc.)

File: src/main/java/com/KacFlor/ShopSpring/service/CardDataService.java (hunk 15 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/service/CardDataService.java]: done

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 8 above)
Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: I expected yes or no answer

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 10 above)
Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: I expected yes or no answer

File: src/main/java/com/KacFlor/ShopSpring/controller/UserController.java (hunk 9 above)
Comment (Ficzer) [reply src/main/java/com/KacFlor/ShopSpring/controller/UserController.java]: I expected yes or no answer

File: src/main/java/com/KacFlor/ShopSpring/service/JwtService.java (hunk 16 above)
Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/service/JwtService.java]: done

**Review by Ficzer — APPROVED** [review]

**Pull request merged**